src/NumericalAlgorithms/DiscontinuousGalerkin/MetricIdentityJacobian.hpp

```diff
@@ -175,12 +175,77 @@ namespace dg {
  * The subtraction technique is most commonly used in finite difference codes.
  */
 template <size_t Dim>
-void metric_identity_jacobian(
+void metric_identity_det_jac_times_inv_jac(
     gsl::not_null<
         InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>*>
         det_jac_times_inverse_jacobian,
     const Mesh<Dim>& mesh,
     const tnsr::I<DataVector, Dim, Frame::Inertial>& inertial_coords,
     const Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
         jacobian) noexcept;
+
+/*!
+ * \ingroup DiscontinuousGalerkinGroup
+ * \brief Compute the Jacobian, inverse Jacobian, and determinant of the
+ * Jacobian so that they satisfy the metric identities.
+ *
+ * Uses `dg::metric_identity_jacobian()` to compute the determinant of the
+ * Jacobian times the inverse Jacobian. Taking the determinant in \f$d\f$
+ * dimensions:
+ *
+ * \f{align}{
+ *
+ *  \mathrm{det}\left(J\frac{\partial \xi^{\hat{\imath}}}{\partial x^i}\right)
+ *  = J^{d-1}.
+ *
+ * \f}
+ *
+ * To get the Jacobian with the correct sign (since in 3d we have computed
+ * \f$J^{2}\f$ and so don't know the sign), we use the following:
+ *
+ * \f{align}{
+ *
+ *  &J =
+ *  &\sqrt[(d-1)]{J^{d-1}}\times
+ *   \left\{
+ *   \begin{array}{ll}
+ *     1 & \mathrm{if} \; d \; \mathrm{even} \\
+ *     \mathrm{sgn}\left(\frac{J^{-1}\partial_\xi x}{\partial_\xi
+ *     x}\right) & \mathrm{if} \; d \; \mathrm{odd}
+ *   \end{array}\right.
+ *
+ * \f}
+ *
+ * We can now compute the inverse Jacobian using:
+ *
+ * \f{align}{
+ *
+ * \frac{\partial \xi^{\hat{\imath}}}{\partial x^i}=
+ *  \frac{1}{J}\left(J\frac{\partial \xi^{\hat{\imath}}}{\partial x^i}\right)
+ *
+ * \f}
+ *
+ * This guarantees that multiplying the determinant of the Jacobian by the
+ * inverse Jacobian gives a result that satisfies the metric identities. We also
+ * compute the Jacobian by inverting the inverse Jacobian, which guarantees they
+ * are (numerical) inverses of each other.
+ *
+ * \warning on entry `jacobian` must be the Jacobian to use for computing the
+ * determinant of the Jacobian times the inverse Jacobian so that it satisfies
+ * the metric identities. On output the `jacobian` is the inverse of the inverse
+ * Jacobian that satisfies the metric identities. The `jacobian` can be computed
+ * analytically or numerically, either is fine.
```

Comment:
I suggest swapping the order of the last two sentences (i.e., move this last sentence into the middle of the paragraph). I think this will be helpful because it will show that the input can be computed in different ways as soon as you've presented the input.